### ISSUE TITLE
Required routing to support Docker containers in Hack Oregon infrastructure

### ISSUE THREAD

**Issue by MikeTheCanuck** [hackoregon/civic-devops#147]
## TL;DR What's the requirement?
- all API containers will have their Django APIs published via the URL http://service.civicpdx.org
- each container will have its own route e.g. Housing Affordability's API container will have all its requests routed to http://service.civicpdx.org/transportation-systems/
- the standard naming of this top-level route is according to the naming convention in #1 
- all resources will be addressed through sub-routes below this  e.g.
    - docs are found at http://service.civicpdx.org/transportation-systems/docs/
    - static files (swagger) are found at http://service.civicpdx.org/transportation-systems/static/
    - API endpoints such as /biketown are found at http://service.civicpdx.org/transportation-systems/biketown/

## Example of compliant routing
- Check out Transportation-Systems' API at http://service.civicpdx.org/transportation-systems/
- their API-wide routing is defined in `urls.py` [here](https://github.com/hackoregon/transportation-systems-backend-2018/blob/d0c1ff811d89ed4e267e638062476493f9ba90a3/transportation_systems_2018/urls.py#L12)

## Why is this necessary?
Good question, probably-crappy answer incoming: it works.  We've hosted API with their endpoints and swagger interfaces like this for two years now.  Are there better/different solutions?  Probably.

We're dependent on DNS for the URL service.civicpdx.org, ALB (application load balancing) for routing requests to each container, CloudFormation for ensuring there are multiple healthy containers available, and the AWS Health Check protocol for ensuring we're not sending requests to a dead container.  Tugging at this solution means re-hydrating the expertise to wrangle all these pieces, and sadly we were understaffed for that amount of work.

**Comment by bhgrant8** [comment]
Two points:

> static files (swagger) are found at http://service.civicpdx.org/transportation-systems/static/

* To clarify on this point, the `<PROJECT_NAME>/static/` route will only appear in the [Django Settings](https://github.com/hackoregon/transportation-systems-backend-2018/blob/80c66df86b1be09f59133d0edb0efe70f0a3fa09/transportation_systems_2018/settings.py#L329). The Swagger and other staticfiles will actually be accessible based on the routing defined in the `urls.py`. 

> We should directly define what should be accessible at the project root, ie: http://service.civicpdx.org/transportation-systems/

We will want something here, and need it for the ALB health checks but

Do we want Swagger at the base or the built in browsable api such as in:
http://service.civicpdx.org/disaster-resilience/

**Comment by MikeTheCanuck** [comment]
Better explanation of the reason for these non-obvious routing changes here:
https://github.com/hackoregon/elections-2018-backend/issues/3

**Comment by MikeTheCanuck** [comment]
> Do we want Swagger at the base or the built in browsable api such as in (old version of Disaster)?

I have no strong opinion here @bhgrant8 - just want something that will respond healthily and can be used by the public.

**Comment by MikeTheCanuck** [comment]
Looks like everyone has decided on their API routing in their containers, and I don't have the energy to refactor to a more consistent pattern across the APIs.
